core/db/migrate/20210906065456_add_unique_stock_item_stock_location_variant_backord_index.rb

```diff
@@ -0,0 +1,5 @@
+class AddUniqueStockItemStockLocationVariantBackordIndex < ActiveRecord::Migration[5.2]
+  def change
+    add_index :spree_stock_items, [:stock_location_id, :variant_id, :backorderable], name: 'stock_item_by_loc_var_id_and_backorderable', unique: true
```

Comment:
```suggestion
    add_index :spree_stock_items, [:stock_location_id, :variant_id, :deleted_at], name: 'stock_item_by_loc_var_id_and_backorderable', unique: true
```

based on

https://github.com/spree/spree/blob/82c6095a2fc3c2c81fbda989d951d964422d636b/core/app/models/spree/stock_item.rb#L12

Reply:
Shouldn't this validation contain the `variant_id` as well?

Reply:
@sebastian-palma it validates `variant_id` in `stock_location_id deleted_at` scope